src/main/java/uk/gov/digital/ho/hocs/casework/api/ActionDataSuspendService.java

```diff
@@ -0,0 +1,162 @@
+
+    @Autowired
+    public ActionDataSuspendService(CaseDataRepository caseDataRepository, SuspensionRepository suspensionRepository, InfoClient infoClient, AuditClient auditClient, CaseNoteService caseNoteService) {
+        this.caseDataRepository = caseDataRepository;
+        this.suspensionRepository = suspensionRepository;
+        this.infoClient = infoClient;
+        this.auditClient = auditClient;
+        this.caseNoteService = caseNoteService;
+    }
+
+    @Override
+    public String getServiceMapKey() {
+        return "suspensions";
+    }
+
+    @Override
+    public List<ActionDataDto> getAllActionsForCase(UUID caseUUID) {
+        Optional<List<ActionDataSuspension>> maybeSuspensions = suspensionRepository.findAllByCaseDataUuid(caseUUID);
+        if (maybeSuspensions.isEmpty()) return List.of();
+        return maybeSuspensions.get().stream().map(suspension ->
+                        new ActionDataSuspendDto(
+                            suspension.getUuid(),
+                            suspension.getCaseTypeActionUuid(),
+                            suspension.getActionSubtype(),
+                            suspension.getCaseTypeActionLabel(),
+                            suspension.getDateSuspensionApplied(),
+                            suspension.getDateSuspensionRemoved()
+                    )).collect(Collectors.toList());
+    }
+
+    public String suspend(UUID caseUUID, UUID existingStageUuid, ActionDataSuspendDto suspendDto) {
+
+        log.debug("Request to suspend case {} received", caseUUID);
+
+        CaseData caseData = caseDataRepository.findActiveByUuid(caseUUID);
+        if (caseData == null) {
+            throw new ApplicationExceptions.EntityNotFoundException(String.format("Case with id: %s does not exist.", caseUUID), CASE_NOT_FOUND);
+        }
+
+        CaseTypeActionDto caseTypeActionDto = infoClient.getCaseTypeActionByUuid(caseData.getType(), suspendDto.getCaseTypeActionUuid());
+        if (caseTypeActionDto == null) {
+            throw new ApplicationExceptions.EntityNotFoundException(String.format("No Case Type Action found for actionId: %s", suspendDto.getCaseTypeActionUuid()), ACTION_DATA_CREATE_FAILURE);
+        }
+
+        if (hasMaxActiveRequests(caseTypeActionDto, caseUUID)) {
+            String msg = String.format("The maximum number of 'Pending' requests of type: %s already exist for caseId: %s", caseTypeActionDto.getActionLabel(), caseUUID);
+            log.error(msg);
+            throw new HttpClientErrorException(HttpStatus.BAD_REQUEST,msg);
+        }
+
+        // Save the suspension event.
+        ActionDataSuspension suspensionEntity = new ActionDataSuspension(
+                caseTypeActionDto.getUuid(),
+                caseTypeActionDto.getActionSubtype(),
+                caseTypeActionDto.getActionLabel(),
+                caseData.getType(),
+                caseUUID,
+                suspendDto.getDateSuspensionApplied(),
```

Comment:
This field is defined as a timestamp but we only pass a date into it, which means all the records have a time of 00:00:00 set against them which is inaccurate. Would be good to revisit this field to maintain accuracy.

Reply:
good point, since only the date is required, I've altered the DB field to be type `DATE`.